quinn-proto/src/connection.rs

```diff
@@ -910,13 +910,19 @@ where
                 return;
             }
             DataSent { .. } => {
-                self.events.push_back(Event::StreamFinished {
-                    stream: stream_id,
-                    stop_reason,
-                });
                 // No need to hold on to the stop_reason since it's propagated above
-                stream.state = ResetSent { stop_reason: None };
-                self.streams.send_streams -= 1;
+                stream.state = if stopped {
+                    ResetRecvd { stop_reason: None }
```

Comment:
This, and the similar change below, is incorrect. `STOP_SENDING` does not constitute acknowledgement of a reset. See [the Stream States](https://quicwg.org/base-drafts/draft-ietf-quic-transport.html#name-stream-states) section of the transport draft for details.

Reply:
Fixed.